src/main/java/org/broadinstitute/hellbender/tools/walkers/GnarlyGenotyper.java

```diff
@@ -0,0 +1,292 @@
+ *
+ * <p>
+ * This tool is designed to perform joint genotyping on multiple samples pre-called with HaplotypeCaller to produce a
+ * multi-sample callset in a super extra highly scalable manner. In any case, the input samples must possess genotype likelihoods produced by HaplotypeCaller
+ * with `-ERC GVCF` or `-ERC BP_RESOLUTION`.
+ *
+ * <h3>Input</h3>
+ * <p>
+ * A GenomicsDB containing the samples to joint-genotype.
+ * </p>
+ *
+ * <h3>Output</h3>
+ * <p>
+ * A final VCF in which all samples have been jointly genotyped.
+ * </p>
+ *
+ * <h3>Usage example</h3>
+ *
+ * <h4>Perform joint genotyping on a set of GVCFs stored in a GenomicsDB</h4>
+ * <pre>
+ * gatk --javaOptions "-Xmx4g" GnarlyGenotyper \
+ *   -R reference.fasta \
+ *   -V gendb://genomicsdb \
+ *   --onlyOutputCallsStartingInIntervals \
+ *   -O output.vcf
+ * </pre>
+ *
+ * <h3>Caveats</h3>
+ * <p><ul>
+ * <li>This tool does not subset to the best alternate alleles and can return highly, highly multialleic variants (>1000 alts for cohorts in the 10s of thousands). Only
+ * GenomicsDB instances should be used as input for this tool, because GenomicsDB will drop PLs for such sites to avoid excessive vcf size.</li>
+ * <li>To generate all the annotations necessary for VQSR, input variants must include the QUALapprox, VarDP and MQ_DP annotations
+ * </ul></p>
+ *
+ * <h3>Special note on ploidy</h3>
+ * <p>This tool assumes all diploid genotypes.</p>
+ *
+ */
+@CommandLineProgramProperties(summary = "Perform \"quick and dirty\" joint genotyping on one or more samples pre-called with HaplotypeCaller",
+        oneLineSummary = "Perform \"quick and dirty\" joint genotyping on one or more samples pre-called with HaplotypeCaller",
+        programGroup = ShortVariantDiscoveryProgramGroup.class)
+@DocumentedFeature
+@BetaFeature
+public final class GnarlyGenotyper extends VariantWalker {
+
+    private static final OneShotLogger warning = new OneShotLogger(GnarlyGenotyper.class);
+
+    private static final boolean SUMMARIZE_PLs = false;  //for very large numbers of samples, save on space and hail import time by summarizing PLs with genotype quality metrics
+
+    public static final int PIPELINE_MAX_ALT_COUNT = 6;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc="File to which variants should be written", optional=false)
+    private File outputFile;
+
+    @Argument(fullName = "output-database-name", shortName = "output-db",
+            doc="File to which the sites-only annotation database derived from these input samples should be written", optional=true)
+    private String outputDbName = null;
+
+    @ArgumentCollection
+    private GenotypeCalculationArgumentCollection genotypeArgs = new GenotypeCalculationArgumentCollection();
+
+    @Argument(fullName = "keep-all-sites", shortName = "keep-all",
```

Comment:
For arguments like this, we generally recommend omitting the short names entirely, as they can get quite cryptic.